src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+class DecoMeasurement(ECMSMeasurement):
+    """Class implementing deconvolution of EC-MS data"""
+
+    def __intit__(self, name, **kwargs):
+        """initialize a deconvolution EC-MS measurement
+
+        Args:
+            name (str): The name of the measurement"""
+        super().__init__(name, **kwargs)
+
+    def get_partial_current(self,
+        signal_name,
+        kernel_obj,
+        tspan=None,
+        t_bg=None,
+        snr=10):
+        """Returns the deconvoluted partial current for a given signal
+
+        Args:
+            signal_name (str): Name of signal for which deconvolution is to
+                be carried out.
+            kernel_obj (Kernel): Kernel object which contains the mass transport
+                parameters
+            tspan (list): Timespan for which the partial current is returned.
+            t_bg (list): Timespan that corresponds to the background signal.
+            snr (int): signal-to-noise ratio used for Wiener deconvolution.
+            """
+
+
+        t_sig, v_sig = self.get_calib_signal(signal_name,
+            tspan=tspan,
+            t_bg=t_bg)
+
+        kernel = kernel_obj.generate_kernel(dt=t_sig[1]-t_sig[0])
+        kernel = np.hstack((kernel, np.zeros(len(sig) - len(kernel))))
+        H = fft(kernel)
+
+        partial_current = np.real(ifft(
+            fft(v_sig)*np.conj(H)/(H*np.conj(H) + (1/snr)**2)
+            ))
+
+        return t_sig, partial_current
+
+
+    def extract_kernel(self, signal_name, cutoff_pot=0, tspan=None, t_bg=None):
+        """Extracts a Kernel object from a measurement.
+
+        Args:
+            signal_name (str): Signal name from which the kernel/impule
+                response is to be extracted.
+            cutoff_pot (int): Potential which the defines the onset of the
+                impulse. Must be larger than the resting potential before the
+                impulse.
+            tspan(list): Timespan from which the kernel/impulse response is
+                extracted.
+            t_bg (list): Timespan that corresponds to the background signal.
+        """
+        x_curr, y_curr = self.get_current(tspan=tspan)
+        x_pot, y_pot = self.get_potential(tspan=tspan)
+        x_sig, y_sig = self.get_signal(signal_name, tspan=tspan, t_bg=t_bg)
+
+        if signal_name == 'M32':
+            t0 = x_curr[np.argmax(y_pot > cutoff_pot)] #time of impulse
+        elif signal_name == 'M2':
+            t0 = x_curr[np.argmax(y_pot < cutoff_pot)]
+        else:
+            print('mass not found')
+
+        x_sig = x_sig - t0
+
+        y_sig = y_sig[x_sig>0]
+        x_sig = x_sig[x_sig>0]
+
+        y_curr = y_curr[x_curr>t0]
+        x_curr = x_curr[x_curr>t0]
+        y_pot = y_pot[x_pot>t0]
+        x_pot = x_pot[x_pot>t0]
+
+        kernel = Kernel(
+            MS_data = np.array([x_sig, y_sig]),
+            EC_data = np.array([x_curr, y_curr, x_pot, y_pot])
+            )
+
+
+        return kernel
+
+
+
+class Kernel:
+    """
```

Comment:
docstring format. Also, I think the potential of this class is not fulfilled, see my comments under the `gen_kernel` method.